README.md

```diff
@@ -1,2 +1,149 @@
-# leaderboard
-This project aims to make use of gitflow, async functions, webpack and hitting of API's
+# Leaderboard
+
+<a name="readme-top"></a>
+
+# 📗 Table of Contents
+
+- [📖 About the Project](#about-project)
+  - [🛠 Built With](#built-with)
+    - [Tech Stack](#tech-stack)
+    - [Key Features](#key-features)
+- [💻 Getting Started](#getting-started)
+  - [Setup](#setup)
+  - [Prerequisites](#prerequisites)
+  - [Usage](#usage)
+  - [Run tests](#run-tests)
+  - [Deployment](#deployment)
+- [👥 Authors](#authors)
+- [👥 Collaborator](#collaborator)
+- [🔭 Future Features](#future-features)
+- [🤝 Contributing](#contributing)
+- [⭐️ Show your support](#support)
+- [🙏 Acknowledgements](#acknowledgements)
+- [📝 License](#license)
+
+# 📖 [Loeaderboard] <a name="about-project"></a>
+
+**[Leaderboard]** is a project that aims to integrate promises and callback and async functions in the calling in the hitting of an api.
+
+<br>
+
+### Tech Stack <a name="tech-stack"></a>
+
+<details>
+  <summary>Client</summary>
+  <ul>
+    <li><a href="https://reactjs.org/">HTML</a></li>
+    <li><a href="https://reactjs.org/">CSS</a></li>
+    <li><a href="https://reactjs.org/">Linters</a></li>
+    <li><a href="https://reactjs.org/">Javascript</a></li>
+  </ul>
+</details>
+
+### Key Features <a name="key-features"></a>
+
+- **[CSS]**
+- **[JavaScript]**
+- **[Linters]**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 💻 Getting Started <a name="getting-started"></a>
+
+### Prerequisites
+
+In order to run this project, you will need the following:
+
+- Web Browser
+- Code Editor
+- Git
+
+### Setup
+
+To get started, clone this repository to your desired folder:
+
+```shell
+cd my-folder
+git clone https://github.com/Samuelwanza/todolist/.git
+
+cd todolist
+
+npm run start
+```
+
+### Usage:
+
+To explore the tools I have used to create this project, follow these steps:
+
+1. Open the project in Visual Studio Code (VSCode).
+2. Take a look at the project structure and files to understand the implementation.
+3. Review the code, HTML, CSS,webpack ,other modules to see how the project is built.
+
+Feel free to explore and customize the project according to your needs.
+
+### Run tests:
+
+To execute the tests for this project, follow these steps:
+
+1. Open the project in Visual Studio Code (VSCode).
+2. Install the "Live Server" extension from the VSCode marketplace.
+3. Then run npm run start to spawn the development server
+4. This will launch the project in your default web browser and automatically refresh it when changes are made.
+5. Verify that the tests are running successfully and check the console for any error messages or test results.
+
+By following these steps, you can effectively run and validate the tests for the project using Live Server in VSCode.
+
+### Deployment:
+
+Deployment date: 10th June, 2023.
+[Deployed Link](https://samuelwanza.github.io/todolist/dist/)
```

Comment:
- [ ] You are still making use of the deployed link for your todolist project. please kindly change this to leaderboard so users are directly to the correct deployed version :+1: